app/models/route_set.rb

```diff
@@ -81,6 +81,24 @@ def register!
     commit_routes
   end
 
+  def delete!
+    return unless any_routes?
+
+    paths =
```

Comment:
Wonder if this should be extracted into a method.

Reply:
Yeah, the separation between routes, gone_routes and redirects feels quite awkward to me. I've added a paths method to handle this now.